package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -458,74 +470,168 @@ class NCDFReader(base.ReaderBase):
-        if n_atoms is not None and n_atoms != self.n_atoms:
-            raise ValueError("Supplied n_atoms ({0}) != natom from ncdf ({1}). "
-                             "Note: n_atoms can be None and then the ncdf value "
-                             "is used!".format(n_atoms, self.n_atoms))
+        self._verify_units(self.trjfile.variables['time'].units, 'picosecond')
+        self._verify_units(self.trjfile.variables['coordinates'].units,
+                           'angstrom')
+
+        # Check for scale_factor attributes for all data variables and
+        # store this to multiply through later (Issue #2323)
+        self.scale_factors = {'time': 1.0,
+                              'cell_lengths': 1.0,
+                              'cell_angles': 1.0,
+                              'coordinates': 1.0,
+                              'velocities': 1.0,
+                              'forces': 1.0}
+
+        for variable in self.trjfile.variables:
+            if hasattr(self.trjfile.variables[variable], 'scale_factor'):
+                if variable in self.scale_factors:
+                    scale_factor = self.trjfile.variables[variable].scale_factor
+                    self.scale_factors[variable] = scale_factor
+                else:
+                    errmsg = ("scale_factors for variable {0} are "
+                              "not implemented".format(variable))
+                    raise NotImplementedError(errmsg)
 
         self.has_velocities = 'velocities' in self.trjfile.variables
+        if self.has_velocities:
+            self._verify_units(self.trjfile.variables['velocities'].units,
+                               'angstrom/picosecond')
+
         self.has_forces = 'forces' in self.trjfile.variables
+        if self.has_forces:
+            self._verify_units(self.trjfile.variables['forces'].units,
+                               'kilocalorie/mole/angstrom')
 
         self.periodic = 'cell_lengths' in self.trjfile.variables
+        if self.periodic:
+            self._verify_units(self.trjfile.variables['cell_lengths'].units,
+                               'angstrom')
+            # Currently the MDA writer outputs 'degrees' rather than the
+            # singular form 'degree' agreed in the conventions. As discussed
+            # in PR #2326 from MDA 1.0 only and 'degree' will be accepted.
+            cell_angle_units = self.trjfile.variables['cell_angles'].units
+            if (cell_angle_units.decode('utf-8') == 'degrees'):
+                wmsg = ("NCDF trajectory {0} uses units of `degrees` for "
+                        "the `cell_angles` variable instead of `degree`. "
+                        "Support for non-AMBER convention units is now "
+                        "deprecated and will end in MDAnalysis version 1.0. "
+                        "Afterwards, reading this file will raise an error.")
+                warnings.warn(wmsg)
```

Comment:
Needs to be a DeprecationWarning. We have fancy code for deprecation decorators somewhere (git grep...) but here it might just be enough to set the warning type appropriately.

To make the logger output clearer, lead wmsg with "DEPRECATED (1.0):"

Reply:
Added DeprecationWarning category in 
4b12592.